providers/BaseProviders/src/main/java/net/ess3/provider/providers/FixedHeightWorldInfoProvider.java

```diff
@@ -0,0 +1,21 @@
+package net.ess3.provider.providers;
+
+import net.ess3.provider.WorldInfoProvider;
+import org.bukkit.World;
+
+public class FixedHeightWorldInfoProvider implements WorldInfoProvider {
+    @Override
+    public String getDescription() {
+        return "World info provider for fixed world heights";
+    }
+
+    @Override
+    public int getMaxSafeHeight(World world) {
+        return world.getEnvironment() == World.Environment.NETHER ? 127 : 255;
```

Comment:
Max build height is 255 but it is possible to teleport to 256 (the block on top of the topmost block in the world).
```suggestion
        return world.getEnvironment() == World.Environment.NETHER ? 128 : 256;
```